### PR TITLE
[FEAT] Deep-Link to Farm Listings

### PR DESCRIPTION
**Pull request by ShiningMinions**
# Description

This PR implements a deep-linking strategy to go directly to a farm's listings given a magic URL. This feature will help support third-party listing tools (like https://sfl.tools), allowing buyers to be taken directly to a farm's listings at the click of a button.

Fixes + Read More Here: https://github.com/sunflower-land/sunflower-land/issues/1366

Code works in the following way:
1. Enter a URL like the following: `{URL}/#/goblins?viewListingsForLand=1`
2. After game load (and if needed, after auth flow), bypass the splash screen and load directly into goblin town.
3. Auto-display the trader modal, with `Land ID 1`'s current trades.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

https://user-images.githubusercontent.com/103600068/187569166-ee2029f5-48e0-4a16-aaed-03b94931f244.mp4

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by veenhdek** [comment]
🚀 fantastic work! As always!

File: src/features/goblins/trader/tradingPost/Trader.tsx [hunk 1]
```diff
@@ -16,11 +17,27 @@ export const Trader: React.FC = () => {
...
+    openTrader();
+  }, [goblinState, searchParams]);
+
   const openTrader = () => {
     goblinService.send("OPEN_TRADING_POST");
     setShowModal(true);
   };
 
+  const handleModalClose = () => {
+    const modifiedParams = Object.assign({}, searchParams);
```
Comment (c-py) [inline src/features/goblins/trader/tradingPost/Trader.tsx:35]:
```suggestion
    const modifiedParams = new URLSearchParams(searchParams);
```

Awesome PR @ShiningMinions ! 🎉 

It's working really nicely. I did encounter one small issue testing. When closing the trader modal, I got the error below on Firefox:

`Uncaught TypeError: modifiedParams.delete is not a function`.

Replacing `modifiedParams` with `const modifiedParams = new URLSearchParams(searchParams);
` seems to resolve this issue.

**Review by c-py — CHANGES_REQUESTED** [review]

**Comment by ShiningMinions** [comment]
@c-py done :+1:

**Review by c-py — APPROVED** [review]
Looking good mate! Excellent addition!

**Pull request merged**